### PR TITLE
[programgen] Generate custom resource options for resources

### LINKED ISSUE #546: [programgen] generate custom resource options from PCL
### What happened?

At the moment, programgen ignores custom resource options. it should include it when possible

### Steps to reproduce

Compile the following PCL program
```hcl
resource provider "pulumi:providers:aws" {
	region = "us-west-2"
}

resource bucket1 "aws:s3:Bucket" {
	options {
		provider = provider
		dependsOn = [provider]
		protect = true
	}
}
```

### Expected Behavior

Should generate custom resource options for resources

### Actual Behavior

Ignore custom resource options, instead it generates:
```java
public static void stack(Context ctx) {
    var provider = new Provider("provider", ProviderArgs.builder()
        .region("us-west-2")
        .build());

    var bucket1 = new Bucket("bucket1");
}
```

### Versions used

_No response_

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
# Description

Fixes #546

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/codegen/testing/test/testdata/resource-options-pp/MyStack.java [hunk 1]
```diff
@@ -15,7 +16,24 @@ public static void stack(Context ctx) {
...
+                .dependsOn(provider)
+                .build());
+
+        var bucketWithoutArgs = new Bucket("bucketWithoutArgs", BucketArgs.Empty, CustomResourceOptions.builder()
+            .provider(provider)
+            .protect(true)
+            .dependsOn(provider)
+            .ignoreChanges(            
+                bucket,
+                lifecycleRules[0])
```
Comment (t0yv0) [inline pkg/codegen/testing/test/testdata/resource-options-pp/MyStack.java:35]: I worry this is a dangling reference. What does this refer to in the PCL file even? 

Perhaps these need to be string literals like "bucket"?

```
  @Nullable List<String> ignoreChanges,
```

Comment (Zaid-Ajaj) [reply pkg/codegen/testing/test/testdata/resource-options-pp/MyStack.java:35]: > What does this refer to in the PCL file even?

I don't know to be honest 😅 programgen doesn't typecheck PCL programs but assumes they are already valid

Comment (t0yv0) [reply pkg/codegen/testing/test/testdata/resource-options-pp/MyStack.java:35]: Perhaps we need to ask around here, but it's plausible that PCL lets you write the string literals without quotes. In Java we do have to quote. The above code won't be right.

Comment (Zaid-Ajaj) [reply pkg/codegen/testing/test/testdata/resource-options-pp/MyStack.java:35]: You are right! expressions in `ignoreChanges` have to be stringified (asked in slack). That is, unless they are variable references 😓 that makes life extra complicated

**Pull request merged**